### PR TITLE
[wip] Added risky_nodes to ExecutionHit

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?

None

### What changes does this PR make to Grapl? Why?

Adds an optional field to ExecutionHit that, if specified, says which node_keys (among the nodes/edges, also in the message) are high-signal and should be attached to the Risk node.

### How were these changes tested?

that's why this is WIP lol

### CODE REVIEW

File: src/python/engagement-creator/src/engagement-creator.py [hunk 1]
```diff
@@ -330,3 +353,8 @@ def lambda_handler(events: Any, context: Any) -> None:
             LOGGER.error(f"mainloop exception {e}")
             LOGGER.error(traceback.print_exc())
             time.sleep(2)
+
+
+if __name__ == "__main__":
+    if IS_LOCAL:
```
Comment (wimax-grapl) [inline src/python/engagement-creator/src/engagement-creator.py:359]: weird check - does this just not run in prod or what

File: src/python/grapl_analyzerlib/grapl_analyzerlib/execution.py [hunk 2]
```diff
@@ -9,7 +9,22 @@ def __init__(
         node_view: "Accepts",
         risk_score: int,
         lenses: Union[List[str], str],
+        risky_nodes: Optional[Sequence[str]] = None,
     ) -> None:
+        """
+        When an Analyzer finds a risk, its' :py:meth:`~grapl_analyzerlib.analyzer.Analyzer.on_response` method will
```
Comment (wimax-grapl) [inline src/python/grapl_analyzerlib/grapl_analyzerlib/execution.py:15]: remove apostrophe on its

File: src/python/engagement-creator/src/engagement-creator.py [hunk 3]
```diff
@@ -258,7 +280,8 @@ def lambda_handler(events: Any, context: Any) -> None:
                         create_edge(mg_client, lens.uid, "scope", from_uid)
                         create_edge(mg_client, lens.uid, "scope", to_uid)
 
-        for node in nodes:
+        risky_nodes = nodes_to_attach_risk_to(nodes, risky_node_keys)
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/python/engagement-creator/src/engagement-creator.py:283]: `nodes_to_attach_risk_to` does that for us, examining the `risky node keys is empty or null` case

[the remark this answers is not in the document]
Comment (colin-grapl) [reply src/python/engagement-creator/src/engagement-creator.py:283]: No idea how I missed that, cool.

File: src/python/engagement-creator/src/engagement-creator.py (hunk 1 above)
Comment (jgrillo-grapl) [reply src/python/engagement-creator/src/engagement-creator.py:359]: Agree 100% this is weird, I think we can fix it by running lambdas in https://hub.docker.com/r/lambci/lambda/ for local grapl. I plan to test it out in the work I'm doing on analyzer deployments.

Comment (wimax-grapl) [reply src/python/engagement-creator/src/engagement-creator.py]: cool. in the meantime, just keeping this documented :)

**Pull request merged**